opsmngr/snapshot_schedule.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const snapshotScheduleBasePath = "groups/%s/backupConfigs/%s/snapshotSchedule"
+
+// SnapshotScheduleService is an interface for using the Snapshot schedule
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/snapshot-schedule/
+type SnapshotScheduleService interface {
+	Get(context.Context, string, string) (*SnapshotSchedule, *atlas.Response, error)
+	Update(context.Context, string, string, *SnapshotSchedule) (*SnapshotSchedule, *atlas.Response, error)
+}
+
+// SnapshotScheduleServiceOp provides an implementation of the SnapshotScheduleService interface
+type SnapshotScheduleServiceOp service
+
+var _ SnapshotScheduleService = &SnapshotScheduleServiceOp{}
+
+type SnapshotSchedule struct {
+	ClusterID                      string        `json:"clusterId"`
+	GroupID                        string        `json:"groupId"`
+	ReferenceTimeZoneOffset        string        `json:"referenceTimeZoneOffset,omitempty"`
+	DailySnapshotRetentionDays     int           `json:"dailySnapshotRetentionDays,omitempty"`
+	ClusterCheckpointIntervalMin   int           `json:"clusterCheckpointIntervalMin,omitempty"`
```

Comment:
are zero(0) values not allowed? if they are you may need to change these to pointers

Reply:
clusterCheckpointIntervalMin = {15, 30, 60}
dailySnapshotRetentionDays = {1 to 365}
monthlySnapshotRetentionMonths = {1 to 36}
**pointInTimeWindowHours ={doc doesn’t specify} -> needs to be changed
referenceHourOfDay = {0 to 24} -> needs to be changed
referenceMinuteOfHour = {0 to 59} -> needs to be changed**
snapshotIntervalHours = {6, 8, 12, 24}
snapshotRetentionDays = {2 to 5}
weeklySnapshotRetentionWeeks = {1 to 52)